api/src/main/java/com/example/api/service/user/UserService.java

```diff
@@ -147,4 +157,52 @@ public String getProfessorRegisterToken() throws WrongUserTypeException {
         log.info("Professor {} fetch ProfessorRegisterToken", user.getEmail());
         return professorRegisterToken.getToken();
     }
+
+    public void deleteProfessorAccount(String email) throws WrongUserTypeException {
+        User user = getCurrentUser();
+        User professor = userRepo.findUserByEmail(email);
+        userValidator.validateProfessorAccount(professor, email);
+        userValidator.validateProfessorAccount(user, user.getEmail());
+
+        changeUserForActivitiesAndAdditionalPoints(user, professor);
+        userRepo.delete(user);
+    }
+
+    public void deleteStudentAccount() throws WrongUserTypeException {
+        User user = getCurrentUser();
+        userValidator.validateStudentAccount(user, user.getEmail());
+        userRepo.delete(user);
+    }
+
+    private void changeUserForActivitiesAndAdditionalPoints(User from, User to){
+        graphTaskRepo.findAll()
+                .stream()
+                .filter(graphTask -> graphTask.getProfessor() == from)
+                .forEach(graphTask -> graphTask.setProfessor(to));
+        fileTaskRepo.findAll()
+                .stream()
+                .filter(fileTask -> fileTask.getProfessor() == from)
+                .forEach(fileTask -> fileTask.setProfessor(to));
+        surveyRepo.findAll()
+                .stream()
+                .filter(survey -> survey.getProfessor() == from)
+                .forEach(survey -> survey.setProfessor(to));
+        infoRepo.findAll()
+                .stream()
+                .filter(info -> info.getProfessor() == from)
+                .forEach(info -> info.setProfessor(to));
+        additionalPointsRepo.findAll()
+                .stream()
+                .filter(additionalPoint -> additionalPoint.getProfessorEmail().equals(from.getEmail()))
+                .forEach(additionalPoint -> additionalPoint.setProfessorEmail(to.getEmail()));
+    }
+
+    public List<String> getAllProfEmails() {
+        User user = getCurrentUser();
+        return userRepo.findAllByAccountTypeEquals(AccountType.PROFESSOR)
+                .stream()
+                .map(User::getEmail)
+                .filter(email -> !email.equals(user.getEmail()))
```

Comment:
Ja bym chyba to `user.getEmail()` wyciągnął przed tego returna do zmiennej i tę zmienną tutaj podawał. Bo nie jestem pewien czy przypadkiem w tej chwili on nie wykonuje `getEmail` przy każdym porównaniu (każdego jednego elementu). Jeżeli tak jest, to lepiej wyrzucić do zmiennej i `getEmail` wykona się raz, a nie X razy